In [2]:
import pandas as pd
from sklearn.preprocessing import MinMaxScaler

In [3]:
data = pd.read_csv("food_access_research_atlas.csv" )

In [5]:
data.head()

CensusTract    State   County  Urban  POP2010  OHU2010  GroupQuartersFlag  \
0   1001020100  Alabama  Autauga      1     1912      693                  0   
1   1001020200  Alabama  Autauga      1     2170      743                  0   
2   1001020300  Alabama  Autauga      1     3373     1256                  0   
3   1001020400  Alabama  Autauga      1     4386     1722                  0   
4   1001020500  Alabama  Autauga      1    10766     4082                  0   

   NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  TractSeniors  TractWhite  \
0         0  0.000000                  0  ...           221        1622   
1       181  0.083410                  0  ...           214         888   
2         0  0.000000                  0  ...           439        2576   
3         0  0.000000                  0  ...           904        4086   
4       181  0.016812                  0  ...          1126        8666   

   TractBlack  TractAsian  TractNHOPI  TractAIAN  TractOMultir  TractHispanic  \
0         217          14           0         14            45             44   
1        1217           5           0          5            55             75   
2         647          17           5         11           117             87   
3         193          18           4         11            74             85   
4        1437         296           9         48           310            355   

   TractHUNV  TractSNAP  
0         26        112  
1         87        202  
2        108        120  
3         19         82  
4        198        488  

[5 rows x 147 columns]

In [6]:
data.describe()

CensusTract         Urban       POP2010       OHU2010  \
count  7.286400e+04  72864.000000  72864.000000  72864.000000   
mean   2.783045e+10      0.757191   4237.285052   1601.837560   
std    1.581256e+10      0.428783   1972.520112    732.106217   
min    1.001020e+09      0.000000      0.000000      0.000000   
25%    1.212708e+10      1.000000   2883.750000   1102.000000   
50%    2.712979e+10      1.000000   4002.000000   1521.000000   
75%    4.103900e+10      1.000000   5323.000000   2018.000000   
max    5.604595e+10      1.000000  37452.000000  16043.000000   

       GroupQuartersFlag      NUMGQTRS      PCTGQTRS  LILATracts_1And10  \
count       72864.000000  72864.000000  72864.000000       72864.000000   
mean            0.007082    109.619606      0.026961           0.126880   
std             0.083855    442.922669      0.095494           0.332841   
min             0.000000      0.000000      0.000000           0.000000   
25%             0.000000      0.000000      0.000000           0.000000   
50%             0.000000      7.000000      0.001779           0.000000   
75%             0.000000     63.000000      0.015516           0.000000   
max             1.000000  19496.000000      1.000000           1.000000   

       LILATracts_halfAnd10  LILATracts_1And20  ...  TractSeniors  \
count          72864.000000       72864.000000  ...  72864.000000   
mean               0.282115           0.111235  ...    552.645806   
std                0.450032           0.314425  ...    352.988078   
min                0.000000           0.000000  ...      0.000000   
25%                0.000000           0.000000  ...    318.000000   
50%                0.000000           0.000000  ...    495.000000   
75%                1.000000           0.000000  ...    716.000000   
max                1.000000           1.000000  ...  17271.000000   

         TractWhite    TractBlack    TractAsian    TractNHOPI     TractAIAN  \
count  72864.000000  72864.000000  72864.000000  72864.000000  72864.000000   
mean    3068.089386    534.273702    201.392347      7.411246     40.242754   
std     1804.353298    887.806852    435.085612     45.084789    181.024758   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%     1832.000000     42.000000     17.000000      0.000000      7.000000   
50%     2903.000000    158.000000     57.000000      1.000000     15.000000   
75%     4111.000000    606.000000    188.000000      5.000000     33.000000   
max    28983.000000  16804.000000  10485.000000   3491.000000   9009.000000   

       TractOMultir  TractHispanic     TractHUNV     TractSNAP  
count  72864.000000   72864.000000  72864.000000  72864.000000  
mean     385.875618     692.764520    145.356239    207.022206  
std      528.776237    1117.876349    230.201932    187.188807  
min        0.000000       0.000000      0.000000      0.000000  
25%       83.000000      87.000000     37.000000     69.000000  
50%      184.000000     240.000000     84.000000    158.000000  
75%      446.000000     746.000000    172.000000    293.000000  
max     8839.000000   15420.000000   6176.000000   2152.000000  

[8 rows x 145 columns]

# Question 1 

In [11]:
county = pd.DataFrame(data["County"].value_counts())

In [15]:
county.count()

County    1837
dtype: int64

In [18]:
data.groupby("County")["Urban"].value_counts()

County     Urban
0          0        7
Abbeville  0        5
           1        1
Acadia     0        6
           1        6
                   ..
Zapata     1        2
           0        1
Zavala     0        3
           1        1
Ziebach    0        1
Name: Urban, Length: 3187, dtype: int64

In [231]:
states = pd.DataFrame(data["State"].value_counts())

In [232]:
states.count()

State    51
dtype: int64

In [19]:
data[data["County"] == "0"]

CensusTract   State County  Urban  POP2010  OHU2010  GroupQuartersFlag  \
1265   2105000300  Alaska      0      0     2055      872                  0   
1326   2195000200  Alaska      0      0     3815     1599                  0   
1327   2198000100  Alaska      0      0     1685      729                  0   
1328   2198000200  Alaska      0      0     2327      908                  0   
1329   2198000300  Alaska      0      0       87       48                  0   
1330   2198940100  Alaska      0      0     1460      509                  0   
1340   2275000300  Alaska      0      0     2369     1053                  0   

      NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  TractSeniors  TractWhite  \
1265         0  0.000000                  1  ...           275        1006   
1326        43  0.011271                  0  ...           438        2711   
1327        37  0.021958                  1  ...           181        1217   
1328        13  0.005587                  0  ...           227        1358   
1329         0  0.000000                  1  ...            18          79   
1330         0  0.000000                  0  ...           133         145   
1340        19  0.008020                  1  ...           374        1719   

      TractBlack  TractAsian  TractNHOPI  TractAIAN  TractOMultir  \
1265           8          11           1        814           215   
1326          15         100           7        614           368   
1327           5           6           7        351            99   
1328           6          13           3        645           302   
1329           0           0           1          1             6   
1330           6           2          10       1210            87   
1340           4          33           1        384           228   

      TractHispanic  TractHUNV  TractSNAP  
1265             75        239        168  
1326            130        157        128  
1327             29        159        203  
1328             70        125        170  
1329              2          7          0  
1330             26         70        177  
1340             37        136        171  

[7 rows x 147 columns]

In [230]:
pd.DataFrame(data["State"].value_counts())

State
California             8044
Texas                  5258
New York               4907
Florida                4214
Pennsylvania           3218
Illinois               3121
Ohio                   2949
Michigan               2774
North Carolina         2192
New Jersey             2007
Georgia                1965
Virginia               1900
Arizona                1526
Indiana                1508
Tennessee              1497
Massachusetts          1476
Washington             1455
Maryland               1399
Wisconsin              1395
Missouri               1393
Minnesota              1336
Colorado               1249
Alabama                1179
Louisiana              1143
Kentucky               1115
South Carolina         1103
Oklahoma               1046
Connecticut             832
Oregon                  830
Iowa                    825
Kansas                  770
Nevada                  687
Arkansas                686
Mississippi             662
Utah                    588
Nebraska                532
New Mexico              499
West Virginia           484
Maine                   355
Hawaii                  332
Idaho                   298
New Hampshire           295
Montana                 271
Rhode Island            242
South Dakota            222
Delaware                218
North Dakota            205
Vermont                 184
District of Columbia    179
Alaska                  167
Wyoming                 132

In [33]:
data[data["State"] == "District of Columbia" ] 

13901    0
13902    0
13903    0
13904    0
13905    0
        ..
14075    0
14076    0
14077    0
14078    0
14079    0
Name: LILATracts_1And10, Length: 179, dtype: int64

## Question 1.2

In [36]:
data["LILATracts_1And10"]

9245

# Question 2

In [234]:
data["Urban"].value_counts()

1    55172
0    17692
Name: Urban, dtype: int64

75 % of the tracts are urban

In [235]:
55172 / (55172 + 17692)

0.757191480017567

In [37]:
data.groupby("Urban")["LILATracts_1And10"].value_counts()

Urban  LILATracts_1And10
0      0                    16352
       1                     1340
1      0                    47267
       1                     7905
Name: LILATracts_1And10, dtype: int64

### Urban
Out of the 75% urban tracts, 14% are food deserts

In [77]:
(7905/(47267+7905))*100

14.327919959399695

### Rural
Out of the 25% Rural tracts, 7% are food deserts

In [79]:
(1340/(1340+16352))*100

7.57404476599593

In [27]:
data.nunique()

CensusTract      72864
State               51
County            1837
Urban                2
POP2010           9123
                 ...  
TractAIAN          972
TractOMultir      2979
TractHispanic     5256
TractHUNV         1601
TractSNAP         1200
Length: 147, dtype: int64

CensusTract      72864
State               51
County            1837

In [21]:
data[data["State"] == "Alaska"]

CensusTract   State          County  Urban  POP2010  OHU2010  \
1179   2013000100  Alaska  Aleutians East      0     3141      553   
1180   2016000100  Alaska  Aleutians West      0     1185      285   
1181   2016000200  Alaska  Aleutians West      0     4376      927   
1182   2020000101  Alaska       Anchorage      1     5736     1946   
1183   2020000102  Alaska       Anchorage      0     5259     2010   
...           ...     ...             ...    ...      ...      ...   
1341   2282000100  Alaska         Yakutat      0      662      270   
1342   2290000100  Alaska   Yukon-Koyukuk      0     1311      558   
1343   2290000200  Alaska   Yukon-Koyukuk      0     1461      595   
1344   2290000300  Alaska   Yukon-Koyukuk      0     1758      658   
1345   2290000400  Alaska   Yukon-Koyukuk      0     1058      406   

      GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
1179                  0      1726  0.549507                  1  ...   
1180                  0       444  0.374684                  1  ...   
1181                  0      2099  0.479662                  0  ...   
1182                  0       139  0.024233                  0  ...   
1183                  0        16  0.003042                  0  ...   
...                 ...       ...       ...                ...  ...   
1341                  0        18  0.027190                  0  ...   
1342                  0         2  0.001526                  1  ...   
1343                  0        28  0.019165                  1  ...   
1344                  0         0  0.000000                  1  ...   
1345                  0         1  0.000945                  1  ...   

      TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  TractAIAN  \
1179           155         660         219        1130          19        876   
1180            75         289          32         178           7        589   
1181           118        1715         300        1428          96        268   
1182           300        4951          57          73           4        295   
1183           479        4391          48         121          12        286   
...            ...         ...         ...         ...         ...        ...   
1341            64         281           2          27          12        237   
1342           117         185           3           2           3       1076   
1343           172         640           5           4           0        713   
1344           183         198           2           6           0       1495   
1345            97         220           0           2           3        708   

      TractOMultir  TractHispanic  TractHUNV  TractSNAP  
1179           237            385        124         84  
1180            90             60         88         31  
1181           569            666         88         47  
1182           356            209         21         65  
1183           401            201         73        106  
...            ...            ...        ...        ...  
1341           103             17         26         26  
1342            42             13        340        222  
1343            99             27        130        134  
1344            57             14        330        208  
1345           125             12        216        101  

[167 rows x 147 columns]

In [20]:
data.describe()

CensusTract         Urban       POP2010       OHU2010  \
count  7.286400e+04  72864.000000  72864.000000  72864.000000   
mean   2.783045e+10      0.757191   4237.285052   1601.837560   
std    1.581256e+10      0.428783   1972.520112    732.106217   
min    1.001020e+09      0.000000      0.000000      0.000000   
25%    1.212708e+10      1.000000   2883.750000   1102.000000   
50%    2.712979e+10      1.000000   4002.000000   1521.000000   
75%    4.103900e+10      1.000000   5323.000000   2018.000000   
max    5.604595e+10      1.000000  37452.000000  16043.000000   

       GroupQuartersFlag      NUMGQTRS      PCTGQTRS  LILATracts_1And10  \
count       72864.000000  72864.000000  72864.000000       72864.000000   
mean            0.007082    109.619606      0.026961           0.126880   
std             0.083855    442.922669      0.095494           0.332841   
min             0.000000      0.000000      0.000000           0.000000   
25%             0.000000      0.000000      0.000000           0.000000   
50%             0.000000      7.000000      0.001779           0.000000   
75%             0.000000     63.000000      0.015516           0.000000   
max             1.000000  19496.000000      1.000000           1.000000   

       LILATracts_halfAnd10  LILATracts_1And20  ...  TractSeniors  \
count          72864.000000       72864.000000  ...  72864.000000   
mean               0.282115           0.111235  ...    552.645806   
std                0.450032           0.314425  ...    352.988078   
min                0.000000           0.000000  ...      0.000000   
25%                0.000000           0.000000  ...    318.000000   
50%                0.000000           0.000000  ...    495.000000   
75%                1.000000           0.000000  ...    716.000000   
max                1.000000           1.000000  ...  17271.000000   

         TractWhite    TractBlack    TractAsian    TractNHOPI     TractAIAN  \
count  72864.000000  72864.000000  72864.000000  72864.000000  72864.000000   
mean    3068.089386    534.273702    201.392347      7.411246     40.242754   
std     1804.353298    887.806852    435.085612     45.084789    181.024758   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%     1832.000000     42.000000     17.000000      0.000000      7.000000   
50%     2903.000000    158.000000     57.000000      1.000000     15.000000   
75%     4111.000000    606.000000    188.000000      5.000000     33.000000   
max    28983.000000  16804.000000  10485.000000   3491.000000   9009.000000   

       TractOMultir  TractHispanic     TractHUNV     TractSNAP  
count  72864.000000   72864.000000  72864.000000  72864.000000  
mean     385.875618     692.764520    145.356239    207.022206  
std      528.776237    1117.876349    230.201932    187.188807  
min        0.000000       0.000000      0.000000      0.000000  
25%       83.000000      87.000000     37.000000     69.000000  
50%      184.000000     240.000000     84.000000    158.000000  
75%      446.000000     746.000000    172.000000    293.000000  
max     8839.000000   15420.000000   6176.000000   2152.000000  

[8 rows x 145 columns]

# Question 3

In [44]:
data.groupby("GroupQuartersFlag")["LILATracts_1And10"].value_counts()

GroupQuartersFlag  LILATracts_1And10
0                  0                    63215
                   1                     9133
1                  0                      404
                   1                      112
Name: LILATracts_1And10, dtype: int64

In [82]:
# group quarters with food deserts

In [75]:
112/(404 + 112)

0.21705426356589147

In [83]:
# # group quarters without food deserts

In [76]:
9133/(63215 +9133)

0.12623707635318185

In [ ]:
# percentage of group quarters

In [45]:
data["GroupQuartersFlag"].value_counts()

0    72348
1      516
Name: GroupQuartersFlag, dtype: int64

In [85]:
(516/(516 + 72348) ) * 100

0.7081686429512516

The number of food deserts are high for urban group quarters tract

In [59]:
data[(data["GroupQuartersFlag"] ==1) & (data["LILATracts_1And10"]==1)]["Urban"].value_counts()

1    107
0      5
Name: Urban, dtype: int64

In [58]:
47038 + 16289

63327

In [72]:
# Ratio of group quarters in urban and rurla areas

In [61]:
data.groupby("Urban")["GroupQuartersFlag"].value_counts()

Urban  GroupQuartersFlag
0      0                    17619
       1                       73
1      0                    54729
       1                      443
Name: GroupQuartersFlag, dtype: int64

#### Rural
Percentage of rural people in group quarters

In [90]:
73/(17619 +73)

0.0041261587158037535

#### Urban
Percentage of urban people in group quarters

In [91]:
443/(443 + 54729)

0.008029435220764155

### Poverty rate in group quarters

In [97]:
# Poverty rate in group quarters
# poverty rate is high in gq

In [70]:
data.groupby("GroupQuartersFlag")["PovertyRate"].mean()

GroupQuartersFlag
0    16.473564
1    29.659302
Name: PovertyRate, dtype: float64

In [73]:
data.groupby("GroupQuartersFlag")["LowIncomeTracts"].value_counts()

GroupQuartersFlag  LowIncomeTracts
0                  0                  41750
                   1                  30598
1                  1                    272
                   0                    244
Name: LowIncomeTracts, dtype: int64

In [99]:
# Ratio of low income in group quarters

In [98]:
272/(272 + 244)

0.5271317829457365

In [101]:
# Ratio of low income in non- group quarters

In [100]:
30598/ (30598 + 41750)

0.42292806988444737

# Question 4

In [94]:
# Low access is not correlated with low income

In [93]:
data[data["LA1and10"] == 0]["LowIncomeTracts"].value_counts()

0    23712
1    21625
Name: LowIncomeTracts, dtype: int64

In [ ]:
# Low access people are rich, rich people wants to stay outside. we need to multivariate with urban, ethnicity.

In [95]:
data[data["LA1and10"] == 1]["LowIncomeTracts"].value_counts()

0    18282
1     9245
Name: LowIncomeTracts, dtype: int64

In [105]:
data.groupby("LA1and10")["PovertyRate"].mean()

LA1and10
0    17.976803
1    14.244898
Name: PovertyRate, dtype: float64

In [108]:
data.groupby("LA1and20")["PovertyRate"].mean()

LA1and20
0    17.849840
1    14.066872
Name: PovertyRate, dtype: float64

In [111]:
data[data["LA1and10"] == 1]["Urban"].value_counts()

1    24322
0     3205
Name: Urban, dtype: int64

In [122]:
data[data["LowIncomeTracts"] == 0]["Urban"].value_counts()

1    30228
0    11766
Name: Urban, dtype: int64

In [123]:
# rich and urban

In [120]:
30228/ (11766 + 30228) 

0.7198171167309616

In [126]:
# rich and rural

In [127]:
1 - 0.7198171167309616


0.2801828832690384

In [124]:
# poor and urban

In [125]:
24944/(5926 + 24944)

0.8080336896663427

## Question 4.1

In [236]:
## Poverty rate is high in food deserted areas

In [102]:
data.groupby("LILATracts_1And10")["PovertyRate"].mean()

LILATracts_1And10
0    15.125722
1    26.484619
Name: PovertyRate, dtype: float64

In [109]:
data.groupby("LILATracts_1And20")["PovertyRate"].mean()

LILATracts_1And20
0    15.248630
1    27.100259
Name: PovertyRate, dtype: float64

In [103]:
data.groupby("LowIncomeTracts")["PovertyRate"].mean()

LowIncomeTracts
0     8.634436
1    27.357924
Name: PovertyRate, dtype: float64

In [ ]:
## Poverty rate is high in non low access areas

In [104]:
data.groupby("LA1and10")["PovertyRate"].mean()

LA1and10
0    17.976803
1    14.244898
Name: PovertyRate, dtype: float64

In [113]:
data.groupby("LA1and20")["PovertyRate"].mean()

LA1and20
0    17.849840
1    14.066872
Name: PovertyRate, dtype: float64

In [313]:
#data.groupby("State").sort_values(by="PovertyRate",ascending=False)

In [129]:
data['PovertyRate'].quantile(0.75) 

22.7

In [297]:
states_pov = pd.DataFrame(data.groupby("State").mean("PovertyRate")["PovertyRate"])

In [298]:
states_pov.head()

PovertyRate
State                  
Alabama       21.222053
Alaska        11.231737
Arizona       18.488663
Arkansas      20.350146
California    16.523695

### Normalising the povertyrate count 
This is done before picking the high poverty states to have it normalised based on the overall data

In [341]:
scaler = MinMaxScaler()

states_pov_nm = pd.DataFrame(scaler.fit_transform(states_pov), columns=["PovertyRate"], index=states_pov.index)

In [342]:
states_pov_nm

PovertyRate
State                            
Alabama                  0.773809
Alaska                   0.112711
Arizona                  0.592930
Arkansas                 0.716112
California               0.462901
Colorado                 0.263934
Connecticut              0.145297
Delaware                 0.207535
District of Columbia     0.674124
Florida                  0.503553
Georgia                  0.726746
Hawaii                   0.080953
Idaho                    0.458046
Illinois                 0.447909
Indiana                  0.554305
Iowa                     0.249215
Kansas                   0.353504
Kentucky                 0.700390
Louisiana                0.801888
Maine                    0.340430
Maryland                 0.095953
Massachusetts            0.214158
Michigan                 0.606417
Minnesota                0.192734
Mississippi              1.000000
Missouri                 0.518495
Montana                  0.443776
Nebraska                 0.265249
Nevada                   0.422996
New Hampshire            0.000000
New Jersey               0.129966
New Mexico               0.743427
New York                 0.420159
North Carolina           0.576896
North Dakota             0.169167
Ohio                     0.616299
Oklahoma                 0.577896
Oregon                   0.481432
Pennsylvania             0.341488
Rhode Island             0.357834
South Carolina           0.663904
South Dakota             0.355663
Tennessee                0.631038
Texas                    0.595956
Utah                     0.260999
Vermont                  0.188294
Virginia                 0.190344
Washington               0.288830
West Virginia            0.666417
Wisconsin                0.354553
Wyoming                  0.182199

In [356]:
states_high_pov =pd.DataFrame(states_pov_nm[ (states_pov_nm["PovertyRate"] > states_pov_nm['PovertyRate'].quantile(0.85))])

In [357]:
states_high_pov

PovertyRate
State                            
Alabama                  0.773809
Arkansas                 0.716112
District of Columbia     0.674124
Georgia                  0.726746
Kentucky                 0.700390
Louisiana                0.801888
Mississippi              1.000000
New Mexico               0.743427

## The statewise food desert count

In [345]:
states_food_deserts = pd.DataFrame(data[data["LILATracts_1And10"] ==1]["State"].value_counts())

In [346]:
states_food_deserts.head()

State
Texas        1041
Florida       582
California    542
Ohio          447
Georgia       444

### Normalising the food desert count 
This is done before picking only the high poverty states to have it normalised based on the overall data

In [368]:
scaler = MinMaxScaler()

states_food_deserts_nm = pd.DataFrame(scaler.fit_transform(states_food_deserts), columns=["food_desert_rate"], index=states_food_deserts.index)

In [369]:
states_food_deserts_nm.head()

food_desert_rate
Texas               1.000000
Florida             0.556522
California          0.517874
Ohio                0.426087
Georgia             0.423188

#### Picking only the high poverty states from the food deserts

In [362]:
states_high_pov.count()

PovertyRate    8
dtype: int64

In [363]:
states_food_deserts_nm = states_food_deserts_nm.loc[states_high_pov.index]

In [364]:
states_food_deserts_nm

food_desert_rate
State                                 
Alabama                       0.231884
Arkansas                      0.165217
District of Columbia          0.000000
Georgia                       0.423188
Kentucky                      0.120773
Louisiana                     0.235749
Mississippi                   0.194203
New Mexico                    0.124638

In [365]:
states_high_pov

PovertyRate
State                            
Alabama                  0.773809
Arkansas                 0.716112
District of Columbia     0.674124
Georgia                  0.726746
Kentucky                 0.700390
Louisiana                0.801888
Mississippi              1.000000
New Mexico               0.743427

### There is no correlation between poverty rates and food desert status
The states with high poverty don't have high number of food deserts

In [366]:
states_high_pov["PovertyRate"].corr(states_food_deserts_nm["food_desert_rate"])

0.21222172433329037

#### Checking correlation of poverty rate and food desert status of all the states 

In [370]:
scaler = MinMaxScaler()
states_food_deserts_nm = pd.DataFrame(scaler.fit_transform(states_food_deserts), columns=["food_desert_rate"], index=states_food_deserts.index)

#### There is very less correlation between povery rate and food deserts in all states

In [371]:
states_pov_nm["PovertyRate"].corr(states_food_deserts_nm["food_desert_rate"])

0.4098760019748423

In [287]:
data[data["LILATracts_1And10"] ==1]

CensusTract    State   County  Urban  POP2010  OHU2010  \
6       1001020700  Alabama  Autauga      1     2891     1188   
11      1001021100  Alabama  Autauga      0     3320     1306   
17      1003010600  Alabama  Baldwin      1     3325     1186   
37      1003011407  Alabama  Baldwin      1     4799     2161   
40      1003011502  Alabama  Baldwin      1    10534     4330   
...            ...      ...      ...    ...      ...      ...   
72791  56021001100  Wyoming  Laramie      1     3072      794   
72797  56021001502  Wyoming  Laramie      1     4899     2149   
72806  56025000200  Wyoming  Natrona      1     4385     2050   
72828  56029965400  Wyoming     Park      1     5769     2543   
72831  56031959400  Wyoming   Platte      0     6575     2886   

       GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
6                      0        36  0.012452                  1  ...   
11                     0        33  0.009940                  1  ...   
17                     0        14  0.004210                  1  ...   
37                     0         4  0.000834                  1  ...   
40                     0         0  0.000000                  1  ...   
...                  ...       ...       ...                ...  ...   
72791                  0       503  0.163737                  1  ...   
72797                  0        11  0.002245                  1  ...   
72806                  0       187  0.042645                  1  ...   
72828                  0        66  0.011440                  1  ...   
72831                  0       103  0.015665                  1  ...   

       TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  \
6               360        2333         435          13           3   
11              493        1441        1810           2           0   
17              305        1323        1851           3           0   
37             1040        4525          56          36           1   
40             1882        7818        1693         113           8   
...             ...         ...         ...         ...         ...   
72791             3        2431         233          72          10   
72797           684        4281         130          44           5   
72806           528        3631         128          36           1   
72828           975        5494           9          20           7   
72831          1375        6273          25          24           4   

       TractAIAN  TractOMultir  TractHispanic  TractHUNV  TractSNAP  
6             11            96             98        134        218  
11            14            53             39        196        262  
17            30           118             92        156        311  
37            26           155             57         64        143  
40            74           828           1314        245        613  
...          ...           ...            ...        ...        ...  
72791         21           305            347          6          4  
72797         30           409            668        166        355  
72806         90           499            753        339        482  
72828         46           193            202         38        133  
72831         30           219            408         69        273  

[9245 rows x 147 columns]

In [288]:
data["State"].value_counts()

California              8044
Texas                   5258
New York                4907
Florida                 4214
Pennsylvania            3218
Illinois                3121
Ohio                    2949
Michigan                2774
North Carolina          2192
New Jersey              2007
Georgia                 1965
Virginia                1900
Arizona                 1526
Indiana                 1508
Tennessee               1497
Massachusetts           1476
Washington              1455
Maryland                1399
Wisconsin               1395
Missouri                1393
Minnesota               1336
Colorado                1249
Alabama                 1179
Louisiana               1143
Kentucky                1115
South Carolina          1103
Oklahoma                1046
Connecticut              832
Oregon                   830
Iowa                     825
Kansas                   770
Nevada                   687
Arkansas                 686
Mississippi              662
Utah          

## Countywise poverty and food desert relationship

In [372]:
county_pov = pd.DataFrame(data.groupby("County").mean("PovertyRate")["PovertyRate"])

In [373]:
county_pov.head()

PovertyRate
County                
0            17.471429
Abbeville    22.250000
Acadia       19.758333
Accomack     24.090909
Ada          14.003390

### Normalising the povertyrate count 
This is done before picking the high poverty states to have it normalised based on the overall data

In [374]:
scaler = MinMaxScaler()

county_pov_nm = pd.DataFrame(scaler.fit_transform(county_pov), columns=["PovertyRate"], index= county_pov.index)

In [375]:
county_pov_nm

PovertyRate
County                    
0                 0.374121
Abbeville         0.476445
Acadia            0.423091
Accomack          0.515865
Ada               0.299858
...                    ...
Yukon-Koyukuk     0.516060
Yuma              0.399902
Zapata            0.766595
Zavala            0.707173
Ziebach           0.862955

[1837 rows x 1 columns]

In [376]:
county_high_pov =pd.DataFrame(county_pov_nm[ ( county_pov_nm["PovertyRate"] > county_pov_nm['PovertyRate'].quantile(0.85))])

In [377]:
county_high_pov

PovertyRate
County                    
Accomack          0.515865
Adair             0.511680
Alachua           0.591198
Alamosa           0.532655
Alexander         0.534164
...                    ...
Yuba              0.512848
Yukon-Koyukuk     0.516060
Zapata            0.766595
Zavala            0.707173
Ziebach           0.862955

[276 rows x 1 columns]

### The County wise food desert count

In [397]:
county_food_deserts = pd.DataFrame(data.groupby("County")["LILATracts_1And10"].sum())

In [398]:
county_food_deserts.head()

LILATracts_1And10
County                      
0                          4
Abbeville                  1
Acadia                     4
Accomack                   1
Ada                        2

#### Normalising the food desert count 
This is done before picking only the high poverty states to have it normalised based on the overall data

In [399]:
scaler = MinMaxScaler()

county_food_deserts_nm = pd.DataFrame(scaler.fit_transform(county_food_deserts), columns=["food_desert_rate"], index=county_food_deserts.index)

In [400]:
county_food_deserts_nm.head()

food_desert_rate
County                     
0                  0.031008
Abbeville          0.007752
Acadia             0.031008
Accomack           0.007752
Ada                0.015504

#### Picking only the high poverty counties from the food deserts

In [401]:
county_high_pov.count()

PovertyRate    276
dtype: int64

In [402]:
county_food_deserts_nm = county_food_deserts_nm.loc[county_high_pov.index]

In [403]:
county_food_deserts_nm

food_desert_rate
County                         
Accomack               0.007752
Adair                  0.023256
Alachua                0.085271
Alamosa                0.023256
Alexander              0.023256
...                         ...
Yuba                   0.031008
Yukon-Koyukuk          0.031008
Zapata                 0.023256
Zavala                 0.023256
Ziebach                0.007752

[276 rows x 1 columns]

In [404]:
county_high_pov

PovertyRate
County                    
Accomack          0.515865
Adair             0.511680
Alachua           0.591198
Alamosa           0.532655
Alexander         0.534164
...                    ...
Yuba              0.512848
Yukon-Koyukuk     0.516060
Zapata            0.766595
Zavala            0.707173
Ziebach           0.862955

[276 rows x 1 columns]

### There is no correlation between poverty rates and food desert status even in countywise
The counties with high poverty don't have high number of food deserts

In [408]:
county_high_pov["PovertyRate"].corr(county_food_deserts_nm["food_desert_rate"])

-0.011332335327506245

#### Checking correlation of poverty rate and food desert status of all the counties 

In [409]:
scaler = MinMaxScaler()
county_food_deserts_nm = pd.DataFrame(scaler.fit_transform(county_food_deserts), columns=["food_desert_rate"], index=county_food_deserts.index)

#### There is very less correlation between povery rate and food deserts in all counties

In [410]:
county_pov_nm["PovertyRate"].corr(county_food_deserts_nm["food_desert_rate"])

0.08831953472995763

# Question 5 

In [411]:
data[( data["LILATracts_1And10"] ==1 ) & ( data["HUNVFlag"] > 0)]

CensusTract    State   County  Urban  POP2010  OHU2010  \
6       1001020700  Alabama  Autauga      1     2891     1188   
11      1001021100  Alabama  Autauga      0     3320     1306   
17      1003010600  Alabama  Baldwin      1     3325     1186   
40      1003011502  Alabama  Baldwin      1    10534     4330   
51      1005950900  Alabama  Barbour      1     4583     1784   
...            ...      ...      ...    ...      ...      ...   
72736  56001963100  Wyoming   Albany      1     7020     3201   
72781  56021000200  Wyoming  Laramie      1     4588     2041   
72790  56021001000  Wyoming  Laramie      1     3026     1479   
72797  56021001502  Wyoming  Laramie      1     4899     2149   
72806  56025000200  Wyoming  Natrona      1     4385     2050   

       GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
6                      0        36  0.012452                  1  ...   
11                     0        33  0.009940                  1  ...   
17                     0        14  0.004210                  1  ...   
40                     0         0  0.000000                  1  ...   
51                     0         0  0.000000                  1  ...   
...                  ...       ...       ...                ...  ...   
72736                  0         0  0.000000                  1  ...   
72781                  0         6  0.001308                  1  ...   
72790                  0         5  0.001652                  1  ...   
72797                  0        11  0.002245                  1  ...   
72806                  0       187  0.042645                  1  ...   

       TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  \
6               360        2333         435          13           3   
11              493        1441        1810           2           0   
17              305        1323        1851           3           0   
40             1882        7818        1693         113           8   
51              574        1842        2519          11          20   
...             ...         ...         ...         ...         ...   
72736           696        6411          96         236           2   
72781           469        3579         253          65           4   
72790           343        2487         135          36          16   
72797           684        4281         130          44           5   
72806           528        3631         128          36           1   

       TractAIAN  TractOMultir  TractHispanic  TractHUNV  TractSNAP  
6             11            96             98        134        218  
11            14            53             39        196        262  
17            30           118             92        156        311  
40            74           828           1314        245        613  
51             1           190            260        182        536  
...          ...           ...            ...        ...        ...  
72736         35           240            444        102         87  
72781         83           604           1325        114        331  
72790         55           297            498        102        246  
72797         30           409            668        166        355  
72806         90           499            753        339        482  

[4832 rows x 147 columns]

In [414]:
data.groupby("LILATracts_1And10")["TractHUNV"].sum()

LILATracts_1And10
0    11482
1     4832
Name: HUNVFlag, dtype: int64

In [ ]:
data

# Question 7a
more work needed

In [220]:
data["LILATracts_1And10"].value_counts()

0    63619
1     9245
Name: LILATracts_1And10, dtype: int64

In [221]:
9245/(9245 + 63619)

0.12688021519543258

In [226]:
data["LILATracts_1And20"].value_counts()

0    64759
1     8105
Name: LILATracts_1And20, dtype: int64

In [229]:
8105/(8105 +64759 )

0.11123462889767237

In [ ]:
data["LILATracts_1And20"].value_counts()

In [227]:
data["LILATracts_halfAnd10"].value_counts()

0    52308
1    20556
Name: LILATracts_halfAnd10, dtype: int64

In [228]:
20556/(20556+52308)

0.28211462450592883

In [ ]:
data[(data["LILATracts_1And10"] ==1) & (data)

# Misc 

In [4]:
data[(data["LA1and10"]==1) & (data["Urban"]== 1)]["lapop1share"]

0        0.709980
2        0.420360
3        0.310868
4        0.245504
5        0.700258
           ...   
72847    0.370982
72849    0.696544
72850    0.999535
72853    0.895222
72863    0.353439
Name: lapop1share, Length: 24322, dtype: float64

In [11]:
data[(data["LA1and10"]==1) & (data["Urban"]== 1) & (data["lapop1share"] < 0.33)]["lapop1share"]

3        0.310868
4        0.245504
50       0.252883
51       0.144556
87       0.221559
           ...   
72793    0.198986
72796    0.233891
72809    0.280915
72832    0.123684
72846    0.178514
Name: lapop1share, Length: 7215, dtype: float64

* find flag for tract where there is less no of supermarkets within 1 miles of access. 
* is LILATracts_1And10 a flag for food desert considering the super market or the accessibility of the tract?
* According to the USDA website LILATracts_1And10 is the flag.
* Low-income and low-access tract measured at 1 mile and 10 miles
Definition: A low-income tract with at least 500 people, or 33 percent of the population, living more than 1 mile (urban areas) or more than 10 miles (rural areas) from the nearest supermarket, supercenter, or large grocery store.

* No vehicle, low access, share of housing units at 1 mile
 **Definition**: Percentage of housing units more than 1 mile from the nearest supermarket, supercenter, or large grocery store without access to a vehicle.
* No vehicle, low access, share of housing units at 10 miles
Definition: Percentage of housing units located more than 10 miles from the nearest supermarket, supercenter, or large grocery store without access to a vehicle. I think this is the variable "lahunv10share"
* Out of the 33% population living outside 1 mile in urban, the above variable will tell us how many didn't have vehicles.
 * lahunv1share - this will tell us Percentage of housing units more than 1 mile from the nearest supermarket, supercenter, or large grocery store without access to a vehicle. But we need to use with filter urban = 1 to calculate only the urban share

* OHU2010 - Occupied housing unit count from 2010 census
* TractHUNV - Total count of housing units without a vehicle in tract 
* If we divide TractHUNV by OHU2010, we should get the share of housing units without vehicles but it does not give us "lahunv10share" as  OHU2010 and TractHUNV don't consider the 1 or 10 miles within supermarket